delivery_price_by_category/models/delivery_price_rule.py

```diff
@@ -19,6 +19,13 @@ class PriceRule(models.Model):
     def _onchange_product_category_price(self):
         self.list_base_price = self.product_category_price
 
+    @api.onchange('product_category_id')
+    def _onchange_product_category_id(self):
+        if self.product_category_id:
+            # Reset the fields when they get hidden in the form
+            self.list_base_price = 0.0
+            self.write(self.default_get(['variable', 'operator', 'max_value']))
```

Comment:
@SimoRubi Are you serious ?

Reply:
@rousseldenis in the method evaluating the rules, I have to call `super` in order not to break inheritance.
`super` evaluates all the rules, then it would evaluate also the _category_ rules, but it would evaluate them based on the fields `'variable', 'operator', 'max_value'`.
This is the only solution that I found that doesn't `return` in a loop nor break inheritance, I'd be happy to find another one though

Reply:
@rousseldenis why can't I do that? Isn't it the same as assigning a variable?
Here I see the `write` call just as a shortcut to
```
self.variable = self.default_get(['variable'])['variable']
self.operator = self.default_get(['operator'])['operator']
self.max_value = self.default_get(['max_value'])['max_value']
```

Reply:
`write` dumps data to DB, which you don't want to do. You should do `update` instead for performing the update on the cache, and eventually being dumped to DB if you click on save button.

Reply:
Fixed up in https://github.com/OCA/delivery-carrier/pull/165/commits/96ec61fd587c50e28aa9996e394d12fea140208c